sktime/annotation/datagen.py

```diff
@@ -8,25 +8,139 @@
+
+    Parameters
+    ----------
+    means : array_like
+        Means of the segments to be generated of shape (L, N)
+    lengths : array_like
+        Lengths of the segments to be generated of shape (L,)
+    vars : float or array_like (default=1.0)
+        Variance of the segments to be generated
+    covs : array_like (default=None)
+        Covariances of segments to be generated of shape (L, N, N)
+        If None, this will be constructed from vars
+    random_state : int or np.random.RandomState
+        Either a random seed or RandomState instance
+
+    Returns
+    -------
+    data : np.array
+        multivariate time series as np.array of shape (X, N)
+        where X = sum(lengths)
+
+    Examples
+    --------
+    >>> from sktime.annotation.datagen import multivariate_shift
+    >>> multivariate_shift([[1, 1], [2, 2], [3, 3]], lengths=[2, 3, 1], random_state=2)
+    array([[ 0.58324215,  0.94373317],
+        [-1.1361961 ,  2.64027081],
+        [ 0.20656441,  1.15825263],
+        [ 2.50288142,  0.75471191],
+        [ 0.94204778,  1.09099239],
+        [ 3.55145404,  5.29220801]])
+
+    >>> from sktime.annotation.datagen import multivariate_shift
+    >>> multivariate_shift([[1, 1], [2, 2], [3, 3]], lengths=[2, 3, 1],
+    vars = [[1.0, 1.0], [1.0, 1.0], [1.0, 1.0]], random_state=2)
+    array([[ 0.58324215,  0.94373317],
+        [-1.1361961 ,  2.64027081],
+        [ 0.20656441,  1.15825263],
+        [ 2.50288142,  0.75471191],
+        [ 0.94204778,  1.09099239],
+        [ 3.55145404,  5.29220801]])
+
+    >>> from sktime.annotation.datagen import multivariate_shift
+    >>> multivariate_shift([[1, 1], [2, 2], [3, 3]], lengths=[2, 3, 1], covs =
+    [[[1.0, 0], [0, 1.0]], [[1.0, 0], [0, 1.0]], [[1.0, 0], [0, 1.0]]], random_state=2)
+    array([[ 0.58324215,  0.94373317],
+        [-1.1361961 ,  2.64027081],
+        [ 0.20656441,  1.15825263],
+        [ 2.50288142,  0.75471191],
+        [ 0.94204778,  1.09099239],
+        [ 3.55145404,  5.29220801]])
+
+    >>> from sktime.annotation.datagen import multivariate_shift
+    >>> multivariate_shift([[1, 3], [4, 5]], lengths=[3, 3], covs = [[[0.5, 0.3],
+    [0.3, 1.0]], [[1.0, 0.3], [0.3, 0.7]]], random_state=2)
+    array([[ 0.78066776,  2.61125356],
+        [ 0.92296736,  0.51689669],
+        [-0.2694238 ,  1.47959507],
+        [ 4.00389069,  3.95225998],
+        [ 5.32264874,  5.05088075],
+        [ 2.62479901,  6.08308546]])
+
+    """
+
+    def get_covs(var):
+        """Fill 1D variance array of length N to 2D covariance array of size (N,N)."""
+        cov = np.zeros((N, N), float)
+        np.fill_diagonal(cov, var)
+        return cov
+
+    L, N = np.array(means).shape
+
+    rng = check_random_state(random_state)
+    assert len(lengths) == L
+
+    # if no covariance is specified, build it from variance
+    # assuming independent random variables
+    if covs is None:
+        assert vars is not None
```

Comment:
I am curious since we default to vars = 1.0 do we need this?  I guess it guards against someone passing `vars= None`

Reply:
yeah you're right, it might be excessive to check, but it's the one thing we need to build up covariance - in terms of debugging I think it's worth leaving it!